### PR TITLE
add device filter option

### PR DESCRIPTION
**Pull request by vdbg**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -244,10 +244,14 @@ def command_unknown(self, update: Update, context: CallbackContext) -> None:
...
     def command_list_devices(self, update: Update, context: CallbackContext) -> None:
         device_groups = self.get_user(update).device_groups
+        device_filter = ' '.join(context.args).lower()
         devices = dict()
         for device_group in device_groups:
             for device in device_group.get_devices().values():
-                devices[device['label']] = device
+                name = device['label']
+                # lower(): Hack because Python doesn't support case-insensitive searches
+                if not device_filter or device_filter in name.lower():
```
Comment (Nickolaim) [inline main.py:253]: Can be simplified to `if device_filter in name.lower():`
```
>>> ' '.join([]).lower()
''
>>> '' in "foo"
True
>>> 'foo' in 'barfoo'
True
```

File: main.py [hunk 2]
```diff
@@ -244,10 +244,14 @@ def command_unknown(self, update: Update, context: CallbackContext) -> None:
...
+                    devices[name] = device
```
Comment (Nickolaim) [inline main.py:254]: This looks very odd.  Usually `device` is something from being iterated over `devices`.  And now after iterating over devices the collection is updated with the device? WTF?!

Consider renaming any of them to something different, so this line is not that confusing.

Comment (vdbg) [reply main.py:254]: I don't understand the comment. `device` is a single device, `devices` is a collection of ... device :)

Let me know what I'm missing.

File: main.py [hunk 3]
```diff
@@ -296,14 +300,14 @@ def configure(self) -> None:
...
+        self.add_command(['dim', 'd'], 'dim device `name` by `number` percent', self.command_dim, params="number name")
         self.add_command(['help', 'h'], 'display help', self.command_help)  # sadly '/?' is not a valid command
-        self.add_command(['info', 'i'], 'get info of device `<name>`', self.command_device_info, params="<name>", isAdmin=True)
-        self.add_command(['list', 'l'], 'list all devices', self.command_list_devices)
-        self.add_command(['on'], 'turn on device `<name>`', self.command_turn_on, params="<name>")
-        self.add_command(['off'], 'turn off device `<name>`', self.command_turn_off, params="<name>")
+        self.add_command(['info', 'i'], 'get info of device `name`', self.command_device_info, params="name", isAdmin=True)
+        self.add_command(['list', 'l'], 'list all devices, or all devices containing `filter`', self.command_list_devices, params="filter")
+        self.add_command(['on'], 'turn on device `name`', self.command_turn_on, params="name")
+        self.add_command(['off'], 'turn off device `name`', self.command_turn_off, params="name")
```
[the remark this answers is not in the document]
Comment (vdbg) [reply main.py:308]: why?

File: main.py (hunk 2 above)
Comment (Nickolaim) [reply main.py:254]: OK, it makes sense now! Plz ignore it.

File: main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nickolaim) [reply main.py:308]: Stuff being in order is better than stuff being out of order?

File: main.py (hunk 1 above)
Comment (vdbg) [reply main.py:253]: ok

File: main.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vdbg) [reply main.py:308]: and .. why is github duping all my replies ;)

[the remark this answers is not in the document]
Comment (vdbg) [reply main.py:308]: ah. on vs. off. k sure.

**Pull request merged**